site-service/src/test/java/uk/ac/ox/ndph/mts/site_service/SiteServiceIntegrationTests.java

```diff
@@ -0,0 +1,88 @@
+package uk.ac.ox.ndph.mts.site_service;
+
+import org.hl7.fhir.r4.model.Organization;
+import org.hl7.fhir.r4.model.ResearchStudy;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import static org.hamcrest.Matchers.containsString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import uk.ac.ox.ndph.mts.site_service.exception.RestException;
+import uk.ac.ox.ndph.mts.site_service.repository.FhirRepository;
+
+@SpringBootTest(properties = { "server.error.include-message=always", "spring.main.allow-bean-definition-overriding=true" })
+@ActiveProfiles("test-all-required")
+@AutoConfigureMockMvc
+class SiteServiceIntegrationTests {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    public FhirRepository repository;
+
+    @Test
+    void TestPostSite_WhenValidInput_Returns201AndId() throws Exception {
+        // Arrange
+        when(repository.saveOrganization(any(Organization.class))).thenReturn("123");
+        when(repository.saveResearchStudy(any(ResearchStudy.class))).thenReturn(any(ResearchStudy.class));
+        
+        String jsonString = "{\"name\": \"name\", \"alias\": \"alias\"}";
+        // Act + Assert
+        this.mockMvc
```

Comment:
You should:
1. Validate that the org doesn't include a site since you didn't provide one.
2. Validate that the org and research study are connected (after you make the logical fix I described above)

Reply:
This will be done later